velox/functions/prestosql/aggregates/PrestoHasher.cpp

```diff
@@ -99,12 +99,53 @@ FOLLY_ALWAYS_INLINE void PrestoHasher::hash<TypeKind::BOOLEAN>(
   });
 }
 
+template <>
+FOLLY_ALWAYS_INLINE void PrestoHasher::hash<TypeKind::BIGINT>(
+    const SelectivityVector& rows,
+    BufferPtr& hashes) {
+  if (vector_->base()->type()->isShortDecimal()) {
+    applyHashFunction(rows, *vector_.get(), hashes, [&](auto row) {
+      return vector_->valueAt<int64_t>(row);
```

Comment:
Decimal values are logical types and are backed by BIGINT. I am not sure if hashing BIGINT based on the logical type is okay.
@kgpai can you take a look at this change? What are the other uses of PrestoHasher?

Reply:
I think this place should be okay, because presto is also implemented in this way. After velxo executes the hash function, it will call [computeHash](https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/aggregates/ChecksumAggregate.cpp#L177C4-L179) to calculate the hash again. Presto has the [same logic](https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/operator/aggregation/ChecksumAggregationFunction.java#L124).